frontend/astra_frontend_asset/src/scenes/SolarSystem/Components/Planet.tsx

```
@@ -0,0 +1,165 @@
+  facingHex = 0x000000,
+  rings = null,
+  ...props
+}, ref) => {
+  const groupRef = useRef<Group>(null!);
+  const planetGroupRef = (ref as React.MutableRefObject<Group>) || useRef<Group>(null!);
+
+  const texture = useLoader(TextureLoader, planetTexture);
+  const planetMaterial = useMemo(() => {
+    if (planetNightTexture) {
+      // planet night texture combined with day texture (night texture is visible when light from the sun is not hitting the planet)
+      const nightMap = useLoader(TextureLoader, planetNightTexture);
```

Biome (2.1.2)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)